opnsense_api/firewall/alias_controller.py

```diff
@@ -1,182 +1,110 @@
-from typing import Union
+from __future__ import annotations
+
+from enum import Enum
+from typing import Union, List
+
+from pydantic import validator, constr
+
+from opnsense_api.util import ProtocolType
+from opnsense_api.util.applicable_item_controller import OPNSenseApplicableItemController
+from opnsense_api.util.exceptions import FailedToParseItemException, InvalidItemException
+from opnsense_api.util.item_controller import OPNSenseItem, TOPNSenseItem
+from opnsense_api.util.parse import parse_selected_enum, parse_selected_keys
+
+LONG_NAME_TO_TYPE_DICT = {
+    "Host(s)": "host",
+    "Network(s)": "network",
+    "Port(s)": "port",
+    "URL (IPs)": "url",
+    "URL Table (IPs)": "urltable",
+    "GeoIP": "geoip",
+    "Network group": "networkgroup",
+    "MAC address": "mac",
+    "BGP ASN": "asn",
+    "Dynamic IPv6 Host": "dynipv6host",
+    "Internal (automatic)": "internal",
+    "External (advanced)": "external"
+}
+
+
+class Alias(OPNSenseItem):
+    name: constr(min_length=1, max_length=32, strip_whitespace=True, regex=r"^[a-zA-Z0-9_]*$")
+    type: constr(to_lower=True)
```

Comment:
Is there a reason we removed it as an enum? I had made it an enum because the number of possible types is fixed and using an enum would make it easier for users of the library to know what alias types were allowed.

Reply:
Yes. Whenever you call searchItem for these, it returns the type as one of the "long names". This allows for easy mapping to the names used during adding and setting. Also, we can use the values to validate any new alias the user creates.